src/internal/generated/custom-css-properties/list.js

```diff
@@ -38,5 +38,6 @@ const customCssPropertiesList = [
   'flashbarStickyBottomMargin',
   'stackedNotificationsBottomMargin',
   'stackedNotificationsDefaultBottomMargin',
+  'columnLayoutColumnCount',
```

Comment:
How about just setting the property inline?

```jsx
<div className={styles['css-grid']} styles={{gridTemplateColumns: `repeat(var(${columnCount}, 1), 1fr)`}}>
```

Custom props are for cross-component sharing. Here the inline styles will work and the solution will be much simpler.

CSS-vars may bring us back old bugs where different column layouts cannot nest properly (not happening in the current PR, but opens a door I would rather not open)

Reply:
It's a good point. I've thought about doing it but I figured the CSS custom props approach would be a bit "nicer/cleaner". Not a strong opinion though.